### PR TITLE
[RW-10488][risk=no] Allow users to delete environments and disks when out of credits

### PR DESCRIPTION
**Pull request by evrii**
Removed disabled panel from apps panel and configuration panels. Show panels as if they are active but disable the create/run buttons.

I have found that the easiest way of testing this is togo into your local db and change the billing_status of the workspace that you want to test with (0=enabled, 1=disabled)

Disabled Apps Panel
image

Disabled Apps Panel with Tooltip
image

Disabled Apps Panel with running Jupyter Environment
image

Disabled Configuration Panel
image

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/configuration-panel.spec.tsx
Comment (evrii) [inline ui/src/app/components/configuration-panel.spec.tsx:1]: These tests were only checking for the presence or absence of the disabled panel, and that is no longer relvant.

File: ui/src/app/components/apps-panel.spec.tsx
Comment (evrii) [inline ui/src/app/components/apps-panel.spec.tsx:1]: Testing to make sure you cannot click on unexpanded apps when billing is inactive.

File: ui/src/app/components/apps-panel.tsx
Comment (evrii) [inline ui/src/app/components/apps-panel.tsx:1]: Disabled clicking on unexpanded apps when billing is inactive.

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 1]
```diff
@@ -107,15 +118,16 @@ const PauseRuntimeButton = (props: { workspace: Workspace }) => {
 };
 
 const JupyterButtonRow = (props: {
```
Comment (evrii) [inline ui/src/app/components/apps-panel/expanded-app.tsx:120]: Disable "Pause and "New Notebook" when billing is inactive.

File: ui/src/app/components/apps-panel.spec.tsx
Comment (evrii) [inline ui/src/app/components/apps-panel.spec.tsx:1]: Testing to make sure you cannot click on unexpanded apps when billing is inactive.

File: ui/src/app/components/apps-panel.tsx
Comment (evrii) [inline ui/src/app/components/apps-panel.tsx:1]: Disabled clicking on unexpanded apps when billing is inactive.

File: ui/src/app/components/apps-panel/expanded-app.tsx
Comment (evrii) [inline ui/src/app/components/apps-panel/expanded-app.tsx:1]: Disable actions that will start environments/apps.

File: ui/src/app/components/gke-app-configuration-panels/create-gke-app-button.tsx [hunk 2]
```diff
@@ -25,9 +32,13 @@ export function CreateGkeAppButton({
   workspaceNamespace,
   onDismiss,
   username,
+  billingStatus,
 }: CreateGKEAppButtonProps) {
   const [creatingApp, setCreatingApp] = useState(false);
-  const createEnabled = !creatingApp && canCreateApp(existingApp);
```
Comment (evrii) [inline ui/src/app/components/gke-app-configuration-panels/create-gke-app-button.tsx:30]: Clearer logic

File: ui/src/app/components/apps-panel/utils.tsx [hunk 3]
```diff
@@ -110,10 +110,6 @@ const isVisible = (status: AppStatus): boolean =>
 export const isAppActive = (app: UserAppEnvironment): boolean =>
   isVisible(app?.status);
 
-// TODO what about ERROR?
-export const canCreateApp = (app: UserAppEnvironment): boolean =>
```
Comment (evrii) [inline ui/src/app/components/apps-panel/utils.tsx:114]: There was only one usage and I replaced that we a clearer usage

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 4]
```diff
@@ -163,14 +170,24 @@ const CromwellButtonRow = (props: {
...
     openAppInIframe(namespace, id, userApp, navigate);
   };
-  const launchButtonDisabled = userApp?.status !== AppStatus.RUNNING;
+  const launchButtonDisabled =
+    billingAccountDisabled || userApp?.status !== AppStatus.RUNNING;
+  let tooltip;
+  if (billingAccountDisabled) {
+    tooltip = BILLING_ACCOUNT_DISABLED_TOOLTIP;
+  } else if (userApp?.status !== AppStatus.RUNNING) {
+    ('Environment must be running to launch RStudio');
```
Comment (Peter-Lavigne) [inline ui/src/app/components/apps-panel/expanded-app.tsx:189]: Should this line assign `tooltip`?

**Review by Peter-Lavigne — CHANGES_REQUESTED** [review]

**Review by Peter-Lavigne — APPROVED** [review]

**Pull request merged**